File: server/apps/api/urls.py
from django.urls import path
from rest_framework import routers
from rest_framework.authtoken import views

from apps.api.views import CategoryViewSet, TransactionViewSet, UserViewSet, CurrencyViewSet, ProfileDetail

router = routers.DefaultRouter()
router.register(r'categories', CategoryViewSet, 'category')
router.register(r'currencies', CurrencyViewSet, 'currency')
router.register(r'transactions', TransactionViewSet, 'transaction')
router.register(r'users', UserViewSet, 'user')

urlpatterns = router.urls + [
    path('profile/', ProfileDetail.as_view()),
    path('token/', views.obtain_auth_token)
]

File: client/src/api/urls.js
const BASE_URL = 'http://localhost:8000/';

export const auth = {
  login: () => `${BASE_URL}api/token/`,
};
export const category = {
  list: () => `${BASE_URL}api/categories/`,
  item: id => `${BASE_URL}api/categories/${id}`,
};
export const currency = {
  list: () => `${BASE_URL}api/currencies/`,
  item: id => `${BASE_URL}api/currencies/${id}`,
};
export const profile = () => `${BASE_URL}api/profile/`;
export const transaction = {
  list: ({ categoryId } = {}) => (
    categoryId
      ? `${BASE_URL}api/transactions/?category=${categoryId}`
      : `${BASE_URL}api/transactions/`
  ),
  item: id => `${BASE_URL}api/transactions/${id}/`,
};
